src/server.c

```diff
@@ -1144,42 +1158,18 @@ void clientsCron(void) {
         c = listNodeValue(head);
         listRotateHeadToTail(server.clients);
 
-        if (c->running_tid != IOTHREAD_MAIN_THREAD_ID &&
-            !(c->running_tid >= start && c->running_tid <= end))
-        {
-            /* Skip clients that are being processed by the IO threads that
-             * are not paused. */
+        /*
+         * Only handle scenarios without iothread or isClientMustHandledByMainThread.
+         * When iothread is enabled, this proc does not operate. Instead, the 
+         * iothread first checks its own clients, followed by the main thread performing 
+         * subsequent processing. This workflow is primarily implemented 
+         * in processClientsFromIOThread and processClientsFromMainThread.
+         */
+        if (server.io_threads_num > 1 && !isClientMustHandledByMainThread(c)) {
             continue;
         }
```

Comment:
let's simply descirbe it here, more details can be provided in IOThreadClientsCron
```suggestion
        /* Clients handled by IO threads will be processed by IOThreadClientsCron. */
        if (c->tid != IOTHREAD_MAIN_THREAD_ID) continue;
```